### PR TITLE
Issue 1

### PR DESCRIPTION
**Pull request by Sabnock01**
This PR is for the first assignment in the Yield mentorship which is to create a simple registry.

### CODE REVIEW

File: src/Registry.sol [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        emit RegisteredName(name, msg.sender);
+    }
+
+    /**
+    @notice Releases a name
+    @param name The name to release
+     */
+    function release(string calldata name) public {
+        require(nameToHolder[name] == msg.sender, "You haven't registered this!");
+        nameToHolder[name] = address(0);
```
Comment (alcueca) [inline src/Registry.sol:33]: I like to use `delete holder[name]`, clearer to see.

File: src/test/Registry.t.sol [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    function testRegisterName() public {
+        vm.expectEmit(false, false, false, true);
+        emit RegisteredName("Sabnock01", address(this));
+        registry.register("Sabnock01");
+    }
+
+    function testReleaseName() public {
+        registry.register("Sabnock01");
```
Comment (alcueca) [inline src/test/Registry.t.sol:28]: As Richie suggested, inheritance is an acceptable way of building state.

**Review by alcueca — CHANGES_REQUESTED** [review]
Minor suggestions on naming and test composition.

File: src/test/lesson_1/Registry.t.sol [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        vm.expectRevert(bytes("You haven't registered this!"));
+        registry.release("alcueca");
+    }
+
+
+    function testCannotRegisterHeldName() public {
+        console.log("Cannot register a name that is taken");
+        registry.register("Sabnock01");
+        vm.expectRevert(bytes("Already registered!"));
+        vm.startPrank(otherSender);
```
Comment (devtooligan) [inline src/test/lesson_1/Registry.t.sol:35]: use prank for one liners.  use startPrank stopPrank for more than one line

**Review by devtooligan — APPROVED** [review]

File: src/test/lesson_1/Registry.t.sol [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    Vm public vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    address otherSender;
+
+    event Registered(address to, string name);
+    event Released(address from, string name);
+
+    function setUp() public virtual {
+        registry = new Registry();
+        otherSender = 0x0000000000000000000000000000000000000001;
```
Comment (devtooligan) [inline src/test/lesson_1/Registry.t.sol:19]: if you use the `label('otherSender')` cheat command, the stack traces will show the string and are easier to follow

Comment (Sabnock01) [reply src/test/lesson_1/Registry.t.sol:19]: `vm.label(otherSender, 'otherSender')` should be the exact usage.

File: src/test/lesson_1/Registry.t.sol (hunk 3 above)
Comment (Sabnock01) [reply src/test/lesson_1/Registry.t.sol:35]: I see.

**Pull request merged**